rlclientlib/extensions/onnx/src/tensor_parser.cc

```diff
@@ -0,0 +1,402 @@
+#include "tensor_parser.h"
+#include "onnx_input.h"
+
+#include <vector>
+#include <string>
+#include <sstream>
+#include <algorithm>
+
+namespace reinforcement_learning { namespace onnx {
+
+namespace tensor_parser
+{
+  enum CONSUME_TYPE : bool { EXCLUSIVE = false, INCLUSIVE = true };
+  enum KNOWN_CHARS : char { END = '\0', DOUBLE_QUOTE = '\"', SEMICOLON = ';', COMMA = ',', OPEN_BRACE = '{', CLOSE_BRACE = '}', BACKSLASH = '\\', COLON = ':' };
+
+  namespace errors
+  {
+    struct error_context
+    {
+    private:
+      std::vector<std::string>& errors;
```

Comment:
nit: See previous comments about ref members and consts

Reply:
Also, can we prefix these variable names with an underscore?

Reply:
I am using references here to ensure that these are always non-null. Would prefer to avoid changing it so I do not need to add nullchecks throughout.